src/Orchard.DisplayManagement/Descriptors/ShapeTemplateStrategy/ShapeTemplateBindingStrategy.cs

```diff
@@ -156,15 +153,16 @@ public void Discover(ShapeTableBuilder builder)
                     if (_logger.IsEnabled(LogLevel.Debug))
                     {
                         _logger.LogDebug("Binding {0} as shape [{1}] for feature {2}",
-                            hit.shapeContext.harvestShapeInfo.TemplateVirtualPath,
+                            hit.shapeContext.harvestShapeInfo.RelativePath,
                             iter.shapeContext.harvestShapeHit.ShapeType,
                             feature.Id);
                     }
+
                     builder.Describe(iter.shapeContext.harvestShapeHit.ShapeType)
                         .From(feature)
                         .BoundAs(
-                            hit.shapeContext.harvestShapeInfo.TemplateVirtualPath,
-                            shapeDescriptor => displayContext => RenderAsync(shapeDescriptor, displayContext, hit.shapeContext.harvestShapeInfo, hit.shapeContext.harvestShapeHit));
+                            hit.shapeContext.harvestShapeInfo.RelativePath, shapeDescriptor => displayContext =>
+                                RenderAsync(hit.shapeContext.harvestShapeInfo.RelativePath, displayContext));
```

Comment:
This will resolve all `IShapeTemplateViewEngine` every time a template is rendered, and also do a Linq query of them to filter the extension. This collection will not change on a per request scope so I think we should:
- Register the `IShapeTemplateViewEngine` implementations in singleton (the `Render` method takes a context so it has access to the scoped container)
- Filter the extension before calling `BoundAs` so that the lambda already know which template view engine to use.